deeplake/core/chunk_engine.py

```diff
@@ -714,34 +722,72 @@ def _samples_to_chunks(
         Returns:
             Tuple[List[BaseChunk], Dict[Any, Any]]
         """
+        extending = start_chunk_row is None and register
+        lengths = None
+        if extending:
+            enc_ids = []
+            enc_count = [0]
+            if self.tensor_meta.htype == "text" and (
+                self.chunk_class != SampleCompressedChunk
+            ):
+                lengths = np.zeros(len(samples), dtype=np.uint32)
+                for i, s in enumerate(samples):
+                    try:
+                        lengths[i] = s.__len__()
+                    except AttributeError:  # None
+                        lengths[i] = 0
+                    except TypeError:  # Numpy scalar str
+                        lengths[i] = str(s).__len__()
```

Comment:
Gettings lengths of strings inside chunks is redundant, so we do it ahead of time in the chunk engine. Using try-except block as its slightly faster than isinstance checks.